src/NodaTime.Test/XmlSchemaTest.XmlSchema.approved.xml

```diff
@@ -0,0 +1,709 @@
+      <xs:pattern value="-?[0-9]{4}-[0-9]{2}-[0-9]{2}(Z|[+-][0-9]{2}(:[0-9]{2}(:[0-9]{2})?)?)" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="OffsetDateTime">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="-?[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]{1,9})?(Z|[+-][0-9]{2}(:[0-9]{2}(:[0-9]{2})?)?)" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="OffsetTime">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]{1,9})?(Z|[+-][0-9]{2}(:[0-9]{2}(:[0-9]{2})?)?)" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="PeriodBuilder">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="P(-?[0-9]+Y)?(-?[0-9]+M)?(-?[0-9]+W)?(-?[0-9]+D)?(T(-?[0-9]+H)?(-?[0-9]+M)?(-?[0-9]+S)?(-?[0-9]+s)?(-?[0-9]+t)?(-?[0-9]+n)?)?" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="YearMonth">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="-?[0-9]{4}-[0-9]{2}" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:complexType name="ZonedDateTime">
+    <xs:simpleContent>
+      <xs:extension base="nodatime:OffsetDateTime">
+        <xs:attribute name="zone" type="nodatime:zone" use="required" />
+        <xs:attribute name="calendar" type="nodatime:calendar" />
+      </xs:extension>
+    </xs:simpleContent>
+  </xs:complexType>
+  <xs:simpleType name="calendar">
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="Badi" />
+      <xs:enumeration value="Coptic" />
+      <xs:enumeration value="Gregorian" />
+      <xs:enumeration value="Hebrew Civil" />
+      <xs:enumeration value="Hebrew Scriptural" />
+      <xs:enumeration value="Hijri Astronomical-Base15" />
+      <xs:enumeration value="Hijri Astronomical-Base16" />
+      <xs:enumeration value="Hijri Astronomical-HabashAlHasib" />
+      <xs:enumeration value="Hijri Astronomical-Indian" />
+      <xs:enumeration value="Hijri Civil-Base15" />
+      <xs:enumeration value="Hijri Civil-Base16" />
+      <xs:enumeration value="Hijri Civil-HabashAlHasib" />
+      <xs:enumeration value="Hijri Civil-Indian" />
+      <xs:enumeration value="ISO" />
+      <xs:enumeration value="Julian" />
+      <xs:enumeration value="Persian Algorithmic" />
+      <xs:enumeration value="Persian Arithmetic" />
+      <xs:enumeration value="Persian Simple" />
+      <xs:enumeration value="Um Al Qura" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="localDate">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="-?[0-9]{4}-[0-9]{2}-[0-9]{2}" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="localDateTime">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="-?[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]{1,9})?" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="zone">
+    <xs:restriction base="xs:string">
```

Comment:
I haven't looked at the code yet, but this makes me nervous. The list of valid time zone IDs changes over time - I don't think it should be part of the schema unless it really has to be. (I wouldn't want an XML document to fail validation with a valid time zone ID just because the ID wasn't known when the schema was created.) Happy to be guided by you on this though.

Reply:
It doesn't *have to be*. You could define a schema with everything being `xs:string` without any restriction but that schema would not be very valuable. But you're absolutely right that rejecting a valid document would be bad. I'll see if it's possible to formally document the list of known valid time zone values while also validating any string, thus accommodating a potential new time zone.

Reply:
I implemented what I think is an elegant solution in d334f28d0a36794a14afa3752754bbeb0d75c39b, using an XML schema union between the enumeration of known zone ids (for documentation purpose) and a string (for validation of future, not yet known time zone ids).